### PR TITLE
Improve CI.

### PR DESCRIPTION
**Pull request by JimiC**
Add build for osx.
Build also against latest stable `node.js`.
Use yarn cache.

### CODE REVIEW

**Comment by JimiC** [comment]
Don't worry. First build for OSX takes a while for Travis Ci to setup the wiring for the project. Every subsequent build start along with the Linux ones and ends in about the same time.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 language: node_js
-node_js: "6"
+os:
+  - linux
+  - osx
+node_js:
+  - "node"
```
Comment (robertohuertasm) [inline .travis.yml]: Why do we use 2 different versions?

Comment (JimiC) [inline .travis.yml]: `node` is for latest current `v7.x` and `6` is for the `LTS`.

Comment (robertohuertasm) [inline .travis.yml]: I know that but I'm not sure that we need to use both checks. I'm using node >7, I don't know you and @jens1o.

Comment (JimiC) [inline .travis.yml]: I use 6.10.x which is the LTS.

Comment (robertohuertasm) [inline .travis.yml]: Ok then 😄 It makes sense.

Comment (JimiC) [inline .travis.yml]: I thought about adding also build for node v4 cause some still use it, but then decided against it. People using node should update to the latest LTS at least.

Comment (robertohuertasm) [inline .travis.yml]: I guess that the only meaningful version is the one that is going to be used when publishing the extension. I wouldn't consider <6 either.

**Comment by robertohuertasm** [comment]
By the way, the `osx` build is taking a lot to complete! 😱

**Comment by JimiC** [comment]
When i first enabled it took 2 hours to initiate. So I suggest you leave this PR last.

**Comment by robertohuertasm** [comment]
I would do so! 2 hours!!

**Pull request merged**